Use existing TCP opts in p0f_impersonate

saturn597 opened the pull request:
Trying to address "Check options in p0f.py" from #399 (the contributions roadmap). Making it so p0f_impersonate uses TCP options from the original packet where possible.

If this commit is on the right track, I can make changes/add tests as needed. Thanks!

p-l- commented on scapy/modules/p0f.py:
```diff
@@ -14,6 +14,7 @@
 import os
 import socket
 import random
+import scapy.modules.six as six
```
You can write `from scapy.modules import six`. Also, since that's a Scapy import, please move it with the other Scapy imports.

p-l- commented on scapy/modules/p0f.py:
```diff
@@ -397,39 +403,70 @@ def p0f_impersonate(pkt, osgenre=None, osdetails=None, signature=None,
             elif opt == 'T':
-                if 'T' in pers[5]:
+                # Determine first timestamp.
+                if uptime is not None:
```
I'm not sure, but shouldn't we always set `ts_a = ts_hint[0]` when `ts_hint[0] and 0 < ts_hint[0] < 2**32`?

saturn597 replied:
Do you mean instead of setting `ts_a` to `uptime` when uptime is set?

To me it seems more intuitive to use the value explicitly passed as an argument (`uptime`), rather than overriding it with the original packet's timestamp. If the caller wanted to use the value in the original packet, I wouldn't expect them to call the function with `uptime=x`.

Also this way the behavior is more consistent with the way it was before these changes. The first timestamp will be equal to uptime if uptime was passed, as before. An alternative if we aren't worried about consistency might be remove the `uptime` argument, since people can just set the timestamp in the original packet.

But for now I've kept it the way I had it. I could definitely be wrong, though, so please let me know if you do prefer it the other way.

saturn597 commented:
Happy to help! Thank you for your review.

I've updated based on your comments, added some tests, and rebased against current master. I also replied back to your comment about setting timestamps. A few things weren't initially working in Python 3. So I fixed the issues I found and added a couple more tests in a second commit (can squash if preferred).

p-l- commented:
Thanks!
